src/intro/mod.rs

```diff
@@ -0,0 +1,92 @@
+extern crate ggez;
+use ggez::*;
+use ggez::timer;
+use ggez::nalgebra;
+use ggez::graphics;
+
+use rand::Rng;
+
+pub struct State {
+    vertigo_angle: f32,
+
+    stars: graphics::Mesh,
+
+    title: graphics::Text
+}
+
+impl State {
+    pub fn new(ctx: &mut Context) -> GameResult<State> {
+        // Generate stars
+        let mut rng = rand::thread_rng();
+
+        let mut mesh_builder = graphics::MeshBuilder::new();
+
+        for _x in 1..=320 {
+            let star_color = graphics::Color::from_rgb(rng.gen_range(0, 10), rng.gen_range(10, 50), rng.gen_range(51, 255));
+
+            mesh_builder.circle(
+                graphics::DrawMode::fill(),
+                nalgebra::Point2::new(rng.gen_range(0.0, 640.0), rng.gen_range(0.0, 640.0)),
+                rng.gen_range(0.0, 3.0),
+                1.0,
+                star_color,
+            );
+        }
+
+        // Text initialization
+        let text = graphics::Text::new("Rusty Gorillas");
+        
+        let s = State {
+            vertigo_angle: 0.0,
+            stars: mesh_builder.build(ctx)?,
+            title: text
+        };
+
+        Ok(s)
+    }
+}
+
+impl event::EventHandler for State {
+
+    fn update(&mut self, ctx: &mut Context) -> GameResult<()> {
+        // Delta time for hardware independent speed
+        let delta = timer::delta(ctx);
+        let delta_millis = (delta.as_secs() as f64 + delta.subsec_millis() as f64 * 0.001) as f32; // TODO: Create helper function for it
+
```

Comment:
I use the calculation of delta as float on 2 places, and I would like to create helper function for it to recycle it on  line 78 as well.

As Rust does not allow other-than-trait methods in `impl` block, how should I address this need correctly?

Reply:
First, I find this line a *bit* fishy. The units of the variable is actually seconds, so the `delta_millis` is highly misleading name. Second, is there a particular reason to first convert everything to f64 and then go back down to f32?

Some suggestions about what you I think you ask here:
* You can create a free standing function that takes `&Context` and returns the `f32`. Or you can put it directly into `ipml State`, but I think it conceptually doesn't belong there.
* If you would prefer it to be a method of either `Context` or `Duration`, you can create a new trait and then implement the trait for that type. That allows „gluing“ methods onto foreign types, but it's seldom worth the effort for simple internal helper function.
* There's [`Duration::as_millis`](https://doc.rust-lang.org/nightly/std/time/struct.Duration.html#method.as_millis), returning `u128`, that can later be casted or the computation done in integers instead of floats.
* There's also [`Duration::as_secs_f32`](https://doc.rust-lang.org/nightly/std/time/struct.Duration.html#method.as_secs_f32). But that one was not yet stabilized, so it requires using nightly and enabling that feature gate. It also risks the fact that unstable features can be removed or changed without much fuzz at any time until they are stabilized, but I think it won't happen in this case.

Reply:
I was probably writing this while half asleep, adressed in 91e26fc7dbb94b4006653c8ac75e52d9dba3b312